Implement Twilio SMS notify MediaUrl support

Chris-Johnston opened the pull request:
## Description:

Adds support for setting the `media_url` parameter of the twilio API client with an [optional attribute under the notify service `data` attribute.](https://www.home-assistant.io/components/notify/#service)

[Per the twilio docs](https://www.twilio.com/docs/sms/send-messages#include-media-in-your-messages) this feature is only available in the US and Canada, for GIF, PNG, or JPEG content.

**Pull request with documentation for [home-assistant.io](https://github.com/home-assistant/home-assistant.io) (if applicable):** home-assistant/home-assistant.io#9792

## Example entry for `configuration.yaml` (if applicable):
```yaml
action:
  - data:
    message: Motion detected.
    data:
      - media_url: http://path.to.your.image/here.jpg
```

## Checklist:
  - [x] The code change is tested and works locally.
  - [ ] Local tests pass with `tox`. **Your PR cannot be merged unless tests pass**
  - [x] There is no commented out code in this PR.
  - [x] I have followed the [development checklist][dev-checklist]

If user exposed functionality or configuration variables are added/changed:
  - [x] Documentation added/updated in [home-assistant.io](https://github.com/home-assistant/home-assistant.io)

If the code communicates with devices, web services, or third-party tools:
  - [ ] [_The manifest file_][manifest-docs] has all fields filled out correctly. Update and include derived files by running `python3 -m script.hassfest`.
  - [ ] New or updated dependencies have been added to `requirements_all.txt` by running `python3 -m script.gen_requirements_all`.
  - [x] Untested files have been added to `.coveragerc`.

If the code does not interact with devices:
  - [ ] Tests have been added to verify that the new code works.

[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html

Chris-Johnston committed:
Implement Twilio SMS notify MediaUrl support

Adds support for setting the `media_url` parameter of the twilio API
client with an optional attribute under the notify `data`
attribute.

Per the twilio docs (https://www.twilio.com/docs/sms/send-messages#include-medi$
this feature is only available in the US and Canada, for
GIF, PNG, or JPEG content.

Chris-Johnston committed:
lint: fix 80 char ruler

homeassistant commented:
Hi @Chris-Johnston, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant#24971).

Once you do that we will be able to review and accept this pull request.

Thanks!

pvizeli reviewed (changes requested):
Use follow style instead object():
```
twillo_args = {
    'message': message,
    'from_number': self.from_... 
}

if ATTR_MEDIAURL in data:
   twillo_args['media_url'] = data[...]

...

self.client.messages.create(to=target, **twillo_args)
```

Chris-Johnston committed:
use kwargs to set the media_url

after testing locally, seems like the previous way of using
object() was not working. this seems to be working

Chris-Johnston commented:
Seems to be working on my machine. Once I got a local test environment set up, the `object()` method was not working, but kwargs seems to be working.

Message w/o image:
```
2019-07-07 15:13:35 INFO (SyncWorker_8) [twilio.http_client] POST Request: https://api.twilio.com/2010-04-01/Accounts/xxxxx/Messages.json
2019-07-07 15:13:35 INFO (SyncWorker_8) [twilio.http_client] PAYLOAD: {'To': '+1xxxxxxxxxx', 'From': '1xxxxxxxxxx', 'Body': 'test'}
2019-07-07 15:13:35 INFO (SyncWorker_8) [twilio.http_client] POST Response: 201 {"sid": "xxxxx", "date_created": "Sun, 07 Jul 2019 22:13:35 +0000", "date_updated": "Sun, 07 Jul 2019 22:13:35 +0000", "date_sent": null, "account_sid": "xxxxx", "to": "+1xxxxxxxxxx", "from": "+1xxxxxxxxxx", "messaging_service_sid": null, "body": "Sent from your Twilio trial account - test", "status": "queued", "num_segments": "1", "num_media": "0", "direction": "outbound-api", "api_version": "2010-04-01", "price": null, "price_unit": "USD", "error_code": null, "error_message": null, "uri": "/2010-04-01/Accounts/xxxxx/Messages/xxxx.json", "subresource_uris": {"media": "/2010-04-01/Accounts/xxx/Messages/xxx/Media.json"}}
```

Message w/ image:
```
2019-07-07 15:14:26 INFO (SyncWorker_3) [twilio.http_client] POST Request: https://api.twilio.com/2010-04-01/Accounts/xxx/Messages.json
2019-07-07 15:14:26 INFO (SyncWorker_3) [twilio.http_client] PAYLOAD: {'To': '+1xxxxxxxxxx', 'From': '1xxxxxxxxxx', 'Body': 'test aaaaaa', 'MediaUrl': 'https://url.goes.here/test.png'}
2019-07-07 15:14:27 INFO (SyncWorker_3) [twilio.http_client] POST Response: 201 {"sid": "xxx", "date_created": "Sun, 07 Jul 2019 22:14:27 +0000", "date_updated": "Sun, 07 Jul 2019 22:14:27 +0000", "date_sent": null, "account_sid": "xxx", "to": "+1xxxxxxxxxx", "from": "+1xxxxxxxxxx", "messaging_service_sid": null, "body": "test aaaaaa", "status": "queued", "num_segments": "1", "num_media": "1", "direction": "outbound-api", "api_version": "2010-04-01", "price": null, "price_unit": "USD", "error_code": null, "error_message": null, "uri": "/2010-04-01/Accounts/xxx/Messages/xxx.json", "subresource_uris": {"media": "/2010-04-01/Accounts/xxx/Messages/xxx/Media.json"}}
```

Chris-Johnston committed:
re-use the ATTR_MEDIAURL attribute
